src/App.tsx

```diff
@@ -1,60 +1,112 @@
-import classNames from 'classnames';
-
-import 'bulma/css/bulma.css';
-import '@fortawesome/fontawesome-free/css/all.css';
-import './App.scss';
+import { useCallback, useEffect, useState } from 'react';
+import cn from 'classnames';
 
 import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
 
-export const App = () => (
-  <main className="section">
-    <div className="container">
-      <div className="tile is-ancestor">
-        <div className="tile is-parent">
-          <div className="tile is-child box is-success">
-            <div className="block">
-              <UserSelector />
-            </div>
+import { User } from './types/User';
+import { Post } from './types/Post';
 
-            <div className="block" data-cy="MainContent">
-              <p data-cy="NoSelectedUser">No user selected</p>
+import { getUsers } from './api/users';
+import { getPosts } from './api/posts';
+import 'bulma/css/bulma.css';
+import '@fortawesome/fontawesome-free/css/all.css';
 
-              <Loader />
+import './App.scss';
 
-              <div
-                className="notification is-danger"
-                data-cy="PostsLoadingError"
-              >
-                Something went wrong!
-              </div>
+export const App = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [userSelected, setUserSelected] = useState<User | null>(null);
+
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState('');
 
-              <div className="notification is-warning" data-cy="NoPostsYet">
-                No posts yet
+  const getUsersPost = useCallback((userId: User['id']) => {
+    setError('');
+    setIsLoading(true);
+
+    getPosts(userId)
+      .then(setPosts)
+      .catch(() => setError('Something went wrong!'))
+      .finally(() => {
+        setIsLoading(false);
+      });
+  }, []);
+
+  useEffect(() => {
+    setError('');
+
+    getUsers()
+      .then(setUsers)
+      .catch(() => setError('Something went wrong!'))
+      .finally(() => {});
```

Comment:
The error handling here is quite generic. It might be beneficial to provide more specific error messages based on the type of error encountered. This will help in debugging and understanding the issue better.